test/RTokenSign.test.js

```diff
@@ -0,0 +1,383 @@
+    }
+
+    // Helper for EIP712 signature.
+    function encodeData(types, primaryType, data) {
+        let encTypes = [];
+        let encValues = [];
+
+        // Add typehash
+        encTypes.push("bytes32");
+        encValues.push(ethUtil.keccak256(encodeType(types, primaryType)));
+
+        // Add field contents
+        for (let field of types[primaryType]) {
+            let value = data[field.name];
+            if (field.type == "string" || field.type == "bytes") {
+                encTypes.push("bytes32");
+                value = ethUtil.keccak256(value);
+                encValues.push(value);
+            } else if (types[field.type] !== undefined) {
+                encTypes.push("bytes32");
+                value = ethUtil.keccak256(encodeData(field.type, value));
+                encValues.push(value);
+            } else if (field.type.lastIndexOf("]") === field.type.length - 1) {
+                throw "TODO: Arrays currently unimplemented in encodeData";
+            } else {
+                encTypes.push(field.type);
+                encValues.push(value);
+            }
+        }
+
+        return abi.rawEncode(encTypes, encValues);
+    }
+
+    // Helper for EIP712 signature.
+    function structHash(types, primaryType, data) {
+        return ethUtil.keccak256(encodeData(types, primaryType, data));
+    }
+
+    before(async () => {
+        console.log("admin is", admin);
+        console.log("bingeBorrower is", bingeBorrower);
+        console.log("business is", business);
+        console.log("holder is", holder);
+    });
+
+    beforeEach(async () => {
+        // Deploys rToken logice.
+        // Mints DAI & rDAI for holder account.
+
+        const chainId = await web3.eth.net.getId();
+        dai_token = await Dai.new(chainId, { from: admin });
+
+        await web3tx(dai_token.mint, "dai_token.mint 1000 -> holder")(holder, toWad(1000), { from: admin });
+
+        {
+            const result = await createCompoundAllocationStrategy(toWad(.1));
+            compoundAS = result.compoundAS;
+        }
+
+        // Deploy the rToken logic/library contract
+        rTokenLogic = await web3tx(RToken.new, "RToken.new")(
+            {
+                from: admin
+            });
+        // Get the init code for rToken
+        const rTokenConstructCode = rTokenLogic.contract.methods.initialize(
+            compoundAS.address,
+            "RToken Test",
+            "RTOKEN",
+            18).encodeABI();
+
+        // Deploy the Proxy, using the init code for rToken
+        const proxy = await web3tx(Proxy.new, "Proxy.new")(
+            rTokenConstructCode, rTokenLogic.address, {
+                from: admin
+            });
+        // Create the rToken object using the proxy address
+        rToken = await RToken.at(proxy.address);
+
+        await web3tx(compoundAS.transferOwnership, "compoundAS.transferOwnership")(rToken.address);
+    });
+
+    it("MintFor Test", async () => {
```

Comment:
Let's add one more negative test case, where for example nonce is invalid, and the mintFor should fail.